app/modules/gpio_pulse.c

```diff
@@ -4,7 +4,7 @@
 #include "platform.h"
 #include "user_interface.h"
 #include "c_types.h"
-#include "c_string.h"
+#include <string.h>
```

Comment:
Seems like this is a redundant include anyway. There are more instances of this type around.
I assume that the pattern based change doesn't detect that aspect - should we use this PR to clean up?

Reply:
I've been trying to find a smart way of weeding out redundant includes, but the best that I've come up with (including are searching SO etc.) is a tight interactive try deleting one and re-make.  If the compile is clean it stays deleted; if it errors then restore it.

As I mention elsewhere we don't do ourselves any favours because the platform abstraction layer doesn't do it's job, IMO.  Your modules still end up including the SDK headers which rather defeats the purpose of abstraction.  Still one step at a time in the right direction is a good thing, I think.